### PR TITLE
tools: Add tests for all discovered functionality in check_format.py

### PR DESCRIPTION
**Pull request by jmarantz**
Signed-off-by: Joshua Marantz <jmarantz@google.com>

*Description*:
>Fills out the testcases for (hopefully) all the formatting issues that check_format.py checks for. In two cases I think the intended behavior was not properly implemented. For one of them I corrected it -- a typo in a table. And in the other one I put in a commented negative test, because the issue should be fixed by the pending refactor in #3496.

*Risk Level*: Low
*Testing*: tools/check_format_test.sh
*Docs Changes*: N/A
*Release Notes*: N/A

### CODE REVIEW

**Comment by mattklein123** [comment]
@danielhochman can you review?

File: tools/check_format_test_helper.py [hunk 1]
```diff
@@ -105,11 +108,15 @@ def checkFileExpectingError(filename, expected_substring):
...
   command, status, stdout = runCheckFormat("check", getInputFile(filename))
   if status != 0:
-    print("status=%d, output:\n" % status)
-    emitStdout(stdout)
+    logging.error("status=%d, output:\n" % status)
+    emitStdoutAsError(stdout)
   return 0
 
 if __name__ == "__main__":
+  parser = argparse.ArgumentParser(description='tester for check_foramt.py.')
```
Comment (danielhochman) [inline tools/check_format_test_helper.py]: nit: spelling

Comment (jmarantz) [reply tools/check_format_test_helper.py]: done

File: tools/check_format_test_helper.py [hunk 2]
```diff
@@ -79,19 +82,19 @@ def fixFileExpectingNoChange(file):
     return 1
   return 0
 
-def emitStdout(stdout):
+def emitStdoutAsError(stdout):
   for line in stdout:
-    print("    %s" % line)
+    logging.error("    %s" % line)
```
Comment (danielhochman) [inline tools/check_format_test_helper.py]: nit: i'd rather this return a newline joined string and then call `logging.error(stdoutAsString(stdout))` inline

**Comment by jmarantz** [comment]
@danielhochman I'm particularly interested in any functional areas of check_format.py that are uncovered. The motivation for this work is to enable refactors in check_format.py that will make it easier to add new functionality, but those refactors are a lot easier to do with confidence if there are comprehensive tests.

**Pull request merged**